jwql/tests/test_utils.py

```diff
@@ -175,6 +177,35 @@ def test_filename_parser(filename, solution):
     assert filename_parser(filename) == solution
 
 
+@pytest.mark.xfail(reason='Known non-compliant files in filesystem')
+def test_filename_parser_whole_filesystem(capsys):
```

Comment:
As far as I can tell, it gives you access to the captured system output (`sysout`?) that would otherwise be handled by `pytest`. It is only here because I was using it earlier to print out the names of the files that were non-compliant, even with the `xfail` being present. So I will remove it now!

Here's where I learned about it: https://stackoverflow.com/a/41533292/9910708